cpp/src/arrow/python/common.h

```diff
@@ -98,6 +98,8 @@ class ARROW_EXPORT OwnedRef {
 // (e.g. if it is released in the middle of a function for performance reasons)
 class ARROW_EXPORT OwnedRefNoGIL : public OwnedRef {
  public:
+  using OwnedRef::OwnedRef;
```

Comment:
I'm not a C++ expert, so I'm curious why this is necessary?

Reply:
Constructors are not inherited by default. I'm not actually using this, so it should cost nothing at runtime. If we ever wanted to construct one of these with a pointer as it's first argument we'd have to define it anyway. I can remove it if you'd like.

Reply:
No problem with me. I had forgotten about non-inheritance of constructors.